### ISSUE TITLE
not:modded still shows Item with 2.0 Mods

### ISSUE THREAD

**Issue by The1CrimsonKing** [DestinyItemManager/DIM#4495]
When I do a filter not:modded stat:total:>60 ,  It still shows armor that has 2.0 Mods equipped. 
When doing is:modded.  it only show armor that has year 1 mods.

**Comment by snowmountainjc** [comment]
I am looking to contribute and it would seem like to me that is missing some small part here in determining what a "mod" in the search-filters.ts file. seems to me this is maybe returning false when it should be returning true for 2.0 mods

**Comment by snowmountainjc** [comment]
ah could 
```
             // enforce that this doesn't have an energy cost (y3 reusables)
              !socket.plug.plugItem.plug.energyCost
```
be the culprit in the hasMod return value. Won't the plugs in d2 items have energy costs?  once i get the environment up and running ill give this a test out.

**Comment by bhollis** [comment]
I think that might be it, yeah!

**Comment by sundevour** [comment]
this is just talking about reverting an intentional change in https://github.com/DestinyItemManager/DIM/commit/206c11fffc1c46a968dec6e6764283352c084962

there is a verbiage problem we need to solve before undoing this:
what should the filter name be for "i've spent some of this armor's energy capacity"?
what should the filter name be for "this armor holds a precious deprecated y2 mod"?

i think they should be different

**Comment by snowmountainjc** [comment]
The goal of this action is to find armor that has not had stats added to it by mods.

Could the option be  to have a stat:base: filter instead that has the ability to just look at the stats on the item regardless before mods? Is that more difficult?

Plus there seems to be some confusion of what "modded" does in these situations for the end user.

**Comment by sundevour** [comment]
already making base stat filters and should be ready this weekend

**Comment by bhollis** [comment]
Sorry I'm a bit confused - wouldn't `is:modded` simply show anything that has a mod in it, regardless of the type of mod?

**Comment by snowmountainjc** [comment]
I would think, from a fresh users eyes, that is:modded or not:modded should be filtering if there are mods in an item or not respectively. 
 
Right now it doesn't seem to be doing that from my testing.

**Comment by sundevour** [comment]
@bhollis the term "mods" is either apples or oranges depending on if it's armor2.0, and i think a filter that requires an accompanying `is:armor2.0` is not a good filter
y2 armor `hasmod` means there's a limited, semi-rare resource (a mod) attached to this item
armor2.0 `hasmod` means "i bothered to click a perk on this"

they're so different that i would like to iron out verbiage and have them be different filters

**Comment by sundevour** [comment]
i think i'd like to split them out into is:modded for y3 and is:hasoldmod for y2

**Comment by The1CrimsonKing** [comment]
As a user i would think the filter should work like this.is:modded or not: modded should show all armor modded or all armor not modded. Then the filter: is:armor show old armor  and is:armor2 show 2.0 armor. So filter would look like not:modded stat:tot>60 Would show all armor not modded with tot stat of more than 60. Or  Is:armor20 not:modded stat:tot>60Shows just 2.0 armor with total stat greater than 60If u think folks want to find just old armor than filter of is:oldarmor could be added. Just thoughts from a user. 
-------- Original message --------From: sundevour <notifications@github.com> Date: 10/19/19  10:36 PM  (GMT-06:00) To: DestinyItemManager/DIM <DIM@noreply.github.com> Cc: The1CrimsonKing <scline19@gmail.com>, Author <author@noreply.github.com> Subject: Re: [DestinyItemManager/DIM] not:modded still shows Item with 2.0 Mods (#4495) i think i'd like to split them out into is:modded for y3 and is:hasoldmod for y2

—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or unsubscribe.
[
{
"@context": "http://schema.org",
"@type": "EmailMessage",
"potentialAction": {
"@type": "ViewAction",
"target": "https://github.com/DestinyItemManager/DIM/issues/4495?email_source=notifications\u0026email_token=ANQ4YV4IMPHAU673MUUGVSDQPPG2FA5CNFSM4JCLYRPKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEBYBR7A#issuecomment-544217340",
"url": "https://github.com/DestinyItemManager/DIM/issues/4495?email_source=notifications\u0026email_token=ANQ4YV4IMPHAU673MUUGVSDQPPG2FA5CNFSM4JCLYRPKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEBYBR7A#issuecomment-544217340",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
"@type": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by snowmountainjc** [comment]
So with the new Stat filters you are making this user would do something similar to `stat:base:` 

And for the future users they would want to see what perks they chose with their armor using new armor mods with `is:modded` and for legacy tracking the user can get the specific y2 mods with `is:hasoldmod`

 That sounds like it makes all the use cases here but maybe should we consider some thing that covers all mods as well? I would have expected `is:modded` to do that so maybe a `is:y3mod` or `is:perkselected` as far as verbiage goes?

**Comment by The1CrimsonKing** [comment]
So how long before the fix is rolled out?

**Comment by delphiactual** [comment]
closed by #4654
